### PR TITLE
docs(k8s): inline environments

### PR DESCRIPTION
**Pull request by Duologic**
Title says it  all.

### CODE REVIEW

File: docs/docs/inline-environments.mdx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+route: "/inline-environments"
+menu: Advanced features
+---
+
+# Inline environments
+
+Inline environments is the practice of defining the environment's config inline
+for evaluation at runtime as opposed to configuring it statically in
+`spec.json`.
+
```
Comment (sh0rez) [inline docs/docs/inline-environments.mdx:12]: As a newcomer to the topic, I'd wish for a section here that gives a rough overview, along those lines:

* `spec.json` is no longer used
* The Jsonnet is expected to hold a `tanka.dev/Environment` object
* This object is expected to hold Kubernetes objects at `.data`

File: docs/docs/inline-environments.mdx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+```jsonnet
+// main.jsonnet
+{
+  some_deployment: {/* ... */ },
+  some_service: {/* ... */ },
+}
+```
+
+
+```js
```
Comment (sh0rez) [inline docs/docs/inline-environments.mdx:41]:
```suggestion
```json
```

**Review by sh0rez — APPROVED** [review]

**Comment by sh0rez** [comment]
Looks good to me, one nit, then :shipit:

File: docs/docs/inline-environments.mdx (hunk 2 above)
Comment (Duologic) [reply docs/docs/inline-environments.mdx:41]: I copy/paste that from your code. :grimacing:

File: docs/docs/inline-environments.mdx (hunk 1 above)
Comment (Duologic) [reply docs/docs/inline-environments.mdx:12]: PTAL, added what you said, simple and clear.

**Pull request merged**